server.py

```diff
@@ -165,14 +165,25 @@ def license():
 if __name__ == '__main__':
 
     if os.path.isfile(certfile) and os.path.isfile(keyfile):
+        ssl_ctx = ssl.create_default_context()
+        ssl_ctx.load_cert_chain(certfile, keyfile)
+
+        # Protocol options: allow TLSv1.1 and later
+        ssl_ctx.options |= ssl.OP_NO_SSLv2
+        ssl_ctx.options |= ssl.OP_NO_SSLv3
+        ssl_ctx.options |= ssl.OP_NO_TLSv1
+
+        # Cipher options: strong ciphers, follow server preferences
+        ssl_ctx.set_ciphers("ECDH+AESGCM128:ECDH+AESGCM256:ECDH+AES128:ECDH+AES256:!aNULL:!MD5:!RC4:!DSS:!EXPORT")
```

Comment:
PS: After some digging for examples, it turns out that the clients that only support HMAC-SHA1 don't support TLSv1.1 anyway, so specifying SHA256 there is fine.